### PR TITLE
(spotify) Working package

### PR DESCRIPTION
**Pull request by Boyan-Kostadinov**

### CODE REVIEW

**Comment by AdmiringWorm** [comment]
while I thank you for your contribution, this is unfortunately not yet done.
It is not up to par with our [contribution guidelines](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines) (Please familiarise yourself with those guidelines)

You will also need to follow our [Package Migration Process](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Package-migration-process) for this to be merged as well.

It would also be preferred to have this package as a automatic update package, with it's own update script.

File: manual/spotify/spotify.nuspec [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
+  <metadata>
+    <id>spotify</id>
+    <version>1.0.49.125</version>
+    <title>Spotify</title>
+    <authors>Spotify</authors>
+    <projectUrl>http://spotify.com</projectUrl>
+    <owners>Boyan Kostadinov (boyank@gmail.com)</owners>
+    <packageSourceUrl>https://github.com/Boyan-Kostadinov/BoxStarter</packageSourceUrl>
```
Comment (AdmiringWorm) [inline manual/spotify/spotify.nuspec]: the package source url should be "https://github.com/chocolatey/chocolatey-coreteampackages/tree/mastermanual/spotify"

File: manual/spotify/spotify.nuspec [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
+  <metadata>
+    <id>spotify</id>
+    <version>1.0.49.125</version>
+    <title>Spotify</title>
+    <authors>Spotify</authors>
+    <projectUrl>http://spotify.com</projectUrl>
+    <owners>Boyan Kostadinov (boyank@gmail.com)</owners>
```
Comment (AdmiringWorm) [inline manual/spotify/spotify.nuspec]: don't include emails in the owners field (I believe that is a requirement on chocolatey).
also, you're missing the users `chocolatey` (should be listed as the first owner), as well as `ferventcoder`

File: manual/spotify/spotify.nuspec [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <packageSourceUrl>https://github.com/Boyan-Kostadinov/BoxStarter</packageSourceUrl>
+    <licenseUrl>https://www.spotify.com/us/legal/end-user-agreement/</licenseUrl>
+    <iconUrl>https://cdn.rawgit.com/ferventcoder/chocolatey-packages/922a763c73805cd98d2c0bb76215720bd77728e6/icons/spotify.png</iconUrl>
```
Comment (AdmiringWorm) [inline manual/spotify/spotify.nuspec]: The icon needs to be located in this repository and in the icons directory, so this iconUrl must be updated

File: manual/spotify/spotify.nuspec [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>Spotify – Millions of tracks, any time you like. Just search for it in Spotify, then play it. Just help yourself to whatever you want, whenever you want it.</description>
```
Comment (AdmiringWorm) [inline manual/spotify/spotify.nuspec]: you're missing a `## Features` list
see [1.2.3 Informative description](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines#123-informative-description)

File: manual/spotify/spotify.nuspec [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    <summary>Spotify is a new way to listen to music</summary>
+    <copyright>Copyright(c) 2017, Spotify Ltd</copyright>
+    <tags>spotify music player radio admin</tags>
```
Comment (AdmiringWorm) [inline manual/spotify/spotify.nuspec]: missing the following tags `freeware`, `cross-platform`
as per [1.2.2 Obligatory tags](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines#122-obligatory-tags)

File: manual/spotify/spotify.nuspec [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    <releaseNotes>
+    Updated to work with the latest requirment for user not to be admin when installing.
+    </releaseNotes>
+  </metadata>
+</package>
```
Comment (AdmiringWorm) [inline manual/spotify/spotify.nuspec]: There should be a empty line at the end of every file
see [2.2 Formatting](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines#22-formatting)

File: manual/spotify/tools/ChocolateyInstall.ps1 [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+# Wait for Spotify to start, then kill it
+$done = $false
+do {
+  if (Get-Process Spotify -ErrorAction SilentlyContinue) {
+    Stop-Process -name Spotify
+    $done = $true
+  }
+
+  Start-Sleep -s 10
+} until ($done)
```
Comment (AdmiringWorm) [inline manual/spotify/tools/ChocolateyInstall.ps1]: There should be a empty line at the end of every file
see [2.2 Formatting](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines#22-formatting)

File: manual/spotify/tools/ChocolateyUnInstall.ps1 [hunk 8]
```diff
@@ -0,0 +1,6 @@
+$packageName = "spotify"
+$fileType = "exe"
+$args = "--uninstall -s"
+$filePath = "$env:APPDATA\Spotify\Spotify.exe"
+ 
+Uninstall-ChocolateyPackage $packageName $fileType $args $filePath
```
Comment (AdmiringWorm) [inline manual/spotify/tools/ChocolateyUnInstall.ps1]: There should be a empty line at the end of every file
see [2.2 Formatting](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines#22-formatting)

**Comment by Boyan-Kostadinov** [comment]
I made all the requested changes, Anything else I need to do?

File: manual/spotify/spotify.nuspec [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+      - Search for it in Spotify, then play it. 
+      - Help yourself to whatever you want, whenever you want it.
+    </description>
+    <summary>Spotify is a new way to listen to music</summary>
+    <copyright>Copyright(c) 2017, Spotify Ltd</copyright>
+    <tags>spotify music player radio admin freeware cross-platform</tags>
+    <releaseNotes>
+    Updated to work with the latest requirment for user not to be admin when installing.
+    </releaseNotes>
+  </metadata>
```
Comment (AdmiringWorm) [inline manual/spotify/spotify.nuspec]: as was requested in a previous comment, please also add `docsUrl` (<https://support.spotify.com/>) and `mailingListUrl` (<https://community.spotify.com/>)

File: manual/spotify/spotify.nuspec [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    <title>Spotify</title>
+    <authors>Spotify</authors>
+    <projectUrl>http://spotify.com</projectUrl>
+    <owners>chocolatey ferventcoder</owners>
+    <packageSourceUrl>https://github.com/chocolatey/chocolatey-coreteampackages/tree/master/manual/spotify</packageSourceUrl>
+    <licenseUrl>https://www.spotify.com/us/legal/end-user-agreement/</licenseUrl>
+    <iconUrl>https://cdn.rawgit.com/chocolatey/chocolatey-coreteampackages/edba4a5849ff756e767cba86641bea97ff5721fe/icons/spotify.png</iconUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>
+      ## Features
```
Comment (AdmiringWorm) [inline manual/spotify/spotify.nuspec]: Please add some more description, not just a list of features.

File: manual/spotify/spotify.nuspec [hunk 11]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
+  <metadata>
+    <id>spotify</id>
+    <version>1.0.49.125</version>
+    <title>Spotify</title>
+    <authors>Spotify</authors>
+    <projectUrl>http://spotify.com</projectUrl>
+    <owners>chocolatey ferventcoder</owners>
```
Comment (AdmiringWorm) [inline manual/spotify/spotify.nuspec]: this should be comma separated, not space separated.

File: manual/spotify/spotify.nuspec [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    <iconUrl>https://cdn.rawgit.com/chocolatey/chocolatey-coreteampackages/edba4a5849ff756e767cba86641bea97ff5721fe/icons/spotify.png</iconUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>
+      ## Features
+      - Millions of tracks, any time you like.
+      - Search for it in Spotify, then play it. 
+      - Help yourself to whatever you want, whenever you want it.
+    </description>
+    <summary>Spotify is a new way to listen to music</summary>
+    <copyright>Copyright(c) 2017, Spotify Ltd</copyright>
```
Comment (AdmiringWorm) [inline manual/spotify/spotify.nuspec]: this should probably be changed to `Copyright © 2017 Spotify AB`, as shown in the software and on the webpage (possibly without the Copyright part)

**Comment by Boyan-Kostadinov** [comment]
What changes does the update script need?

**Comment by AdmiringWorm** [comment]
@Boyan-Kostadinov I found a place were the version can be parsed (unofficial location, and probably a day late or something). I'm changing it to parse the version listed there, and verify the downloaded file is that version.

**Comment by AdmiringWorm** [comment]
I also need to update the install script, which currently fails when running on the test environment (I believe I've found why, so should be updated soon)

**Comment by AdmiringWorm** [comment]
@Boyan-Kostadinov I just updated your PR with some suggestions and some required changes (fixing the scheduling of tasks).

@chocolatey/chocolateycommunitypackages any one available to look over the changes for this PR?

**Comment by Boyan-Kostadinov** [comment]
I had a line in there to calculate the checksum, I think that needs to go into the BeforeUpdsate, either from the downloaded file or with the remote checksum method.

$Latest.Checksum32 = (Get-FileHash $file -Algorithm $checksumType | ForEach Hash).ToLowerInvariant()

**Comment by majkinetor** [comment]
U r using bad solution for safe deleting. U need to disable it on ur local system. Use proxy replacement for rm command otherwise majority of scripts using rm will fail like that.

**Comment by Boyan-Kostadinov** [comment]
Son of a..., I should have known. Yes, I have an alias for "rm," and I use remove-item when I want the official thing..Thanks for pointing that out.

**Comment by AdmiringWorm** [comment]
@pascalberger @majkinetor @gep13 
could one of you please review the this PR?
Since I'm the I made some changes to it, I need someone else to review my changes.

**Comment by majkinetor** [comment]
@Boyan-Kostadinov

Proxy functions keep original arguments,  thats how u do stuff like that. Unfortunatelly most devs suck in Powershell.

**Comment by Boyan-Kostadinov** [comment]
Does the $Latest:ChecksimType32 need to be set somewhere?

**Comment by majkinetor** [comment]
Not generally. Yes if you don't use automatic checksum or get-remotefiles.

File: automatic/spotify/update.ps1 [hunk 13]
```diff
@@ -0,0 +1,43 @@
+Import-Module AU
+
+# We only use this for checking for Spotify versions, nothing else
+$releasesUrl = 'http://www.filehorse.com/download-spotify/'
+$downloadUrl = 'https://download.spotify.com/SpotifyFullSetup.exe'
+
+function global:au_BeforeUpdate {
+  # we need to verify that the downloaded file is the correct version
+  Get-RemoteFiles -Purge
+  $file = Get-ChildItem "$PSScriptRoot\tools" -Filter "*.exe" | select -First 1 | % { Get-Item $_.FullName }
```
Comment (majkinetor) [inline automatic/spotify/update.ps1]: You can do it that way I guess ...
Don't know whats frong with `gi tools\*.exe | select -first 1`

File: automatic/spotify/update.ps1 [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+  [version]$productVersion = $file.VersionInfo.ProductVersion -replace '([\d\.]+)\..*','$1'
+
+  if ($productVersion -gt [version]$Latest.Version) {
+    throw "New version is released, but not yet updated on filehorse"
```
Comment (majkinetor) [inline automatic/spotify/update.ps1]: So, filehorse is authoritative horse for the spotify ?!

File: automatic/spotify/tools/ChocolateyInstall.ps1 [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+    checksumType    = 'sha256'
+    silentArgs      = '/silent'
+    validExitCodes  = @(0, 1641, 3010)
+}
+
+# Download the installer
+$arguments['file'] = Get-ChocolateyWebFile @arguments
+
+# It doesn't matter what time we choose, we need to start it manually
+schtasks.exe /Create /SC Once /st (Get-Date -Format 'HH:mm') /TN $arguments['packageName'] /TR "'$($arguments['file'])' $($arguments['silentArgs'])" /F 2>$null
```
Comment (majkinetor) [inline automatic/spotify/tools/ChocolateyInstall.ps1]: This probably belongs to the helper with something describing the intent. 

Just for reference: https://github.com/chocolatey/chocolatey-coreteampackages/blob/aa05288b177924da94359f27ffc9217541f323cb/automatic/virtualbox/tools/helpers.ps1#L15

File: automatic/spotify/update.ps1 (hunk 13 above)
Comment (AdmiringWorm) [inline automatic/spotify/update.ps1]: My mind just went mostly blank on this line, was the only thing I could remember when I updated the PR.
I'll change it to use `Get-Item` instead though when I merge it.

File: automatic/spotify/update.ps1 (hunk 14 above)
Comment (AdmiringWorm) [inline automatic/spotify/update.ps1]: well, kind of I guess.

File: automatic/spotify/tools/ChocolateyInstall.ps1 (hunk 15 above)
Comment (AdmiringWorm) [inline automatic/spotify/tools/ChocolateyInstall.ps1]: true, it should.
Didn't prioritise it then though, probably should be moved sometime in the future.

**Comment by AdmiringWorm** [comment]
@Boyan-Kostadinov this have been merged, but seems like it don't show as such. (maybe because it can't find the source repo+branch)

**Pull request closed without merging**